### PR TITLE
Hide mark order complete button unpaid orders

### LINKED ISSUE #6894: Hide mark order complete button for unpaid order
As per the discussion [here](https://github.com/woocommerce/woocommerce-android/pull/6802#issuecomment-1171388641), the reasoning for removing the button for unpaid orders is that it's a primary CTA which marks the order as paid, rather than prompting the merchant to take payment. We don't want to confuse people so that they give the product to a customer without taking payment.

The correct CTA for an unpaid order is Collect Payment alone.

For a paid order, the merchant must pick, pack, and send the products out. Once they've done that, they can mark the order complete. Therefore, the `Mark Order Complete` button should be shown on `paid Processing` orders.

### PR DESCRIPTION
**Pull request by rossanafmenezes**
Closes: #6894 

### Description

This PR tweaks the visibility of the `MarkOrderCompleteButton` so it's hidden for all unpaid orders and for orders that already have the COMPLETE status

### Testing instructions

1. Go to the Orders list
2. Open an order that has the COMPLETE status
3. Make sure the Mark Order Complete button is not showing
4. open other orders with different statuses and confirm the button visibility ( if the order is unpaid the button should be hidden is the order is paid - meaning it shows a pay date) the button should be displayed - unless the status is COMPLETE

### Images/gif

 |

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt [hunk 1]
```diff
@@ -1082,6 +1083,52 @@ class OrderDetailViewModelTest : BaseUnitTest() {
             )
         }
 
+    @Test
+    fun `when order is paid, order complete button should be hidden`() =
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt]: ❓ `when order is paid, order complete button should be hidden` Should this test name be `when order is paid, order complete button should be **visible**`?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt [hunk 2]
```diff
@@ -1082,6 +1083,52 @@ class OrderDetailViewModelTest : BaseUnitTest() {
...
+
+            // When
+            viewModel.start()
+
+            // Then
+            assertThat(isMarkOrderCompleteButtonVisible).isTrue
+        }
+
+    @Test
+    fun `when order is not paid, order complete button should be hidden`() =
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt:1133]: ❓ What do you think about writing similar test to verify that the mark order complete button is hidden for order with status as "Complete"?

`when order status is completed, order complete button should be hidden`

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt [hunk 3]
```diff
@@ -1082,6 +1083,52 @@ class OrderDetailViewModelTest : BaseUnitTest() {
...
+        testBlocking {
+            // Given
+            val orderStatusStub = Order.Status.fromDataModel(CoreOrderStatus.PROCESSING)
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt:1136]: ❓ What do you think about writing a similar test to verify that when the order is not paid, mark order complete button should be hidden for different order states? For example: `on hold`, `Failed`, `Canceled` ...etc

**Review by AnirudhBhat — COMMENTED** [review]
Great job @rossanafmenezes I've tested the changes and it works as expected!

I've left a few minor comments related to unit tests. Basically, I think we can add a few more tests for this change. Let me know your thoughts on it.

Again, Great job 👏

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt (hunk 1 above)
Comment (rossanafmenezes) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt]: 100% right. Sorry about that. fixed

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt (hunk 2 above)
Comment (rossanafmenezes) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt:1133]: ✅ Added. Thanks

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt (hunk 3 above)
Comment (rossanafmenezes) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt:1136]: Absolutely! Thank you for the tip. Just added those

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt]: No reason at all to apologize.

**Review by AnirudhBhat — APPROVED** [review]
LGTM! Thank you so much!

**Pull request merged**